internal/jobs/common.go

```diff
@@ -31,25 +38,26 @@ func finishWithSuccess(ctx context.Context, reservationId int64) {
 		ctx = copyContext(ctx)
 	}
 
+	// get details
 	rDao := dao.GetReservationDao(ctx)
 	reservation, err := rDao.GetById(ctx, reservationId)
 	if err != nil {
 		logger.Warn().Err(err).Msg("unable to update job status: get by id")
 		return
 	}
-	logger.Debug().Msgf("Job step: %d/%d", reservation.Step, reservation.Steps)
+	if reservation.Step == reservation.Steps {
+		logger.Info().Msgf("Finishing reservation with success at step %d/%d", reservation.Step, reservation.Steps)
+	} else {
+		logger.Error().Msgf("Finishing reservation with success at step %d/%d", reservation.Step, reservation.Steps)
+	}
 
 	// total count of reservations
 	metrics.IncReservationCount(reservation.Provider.String(), "success")
 
-	// if this was the last step, set the success flag
-	if reservation.Step >= reservation.Steps {
-		logger.Info().Msgf("All jobs executed, marking job as success")
-
-		err = rDao.FinishWithSuccess(ctx, reservationId)
-		if err != nil {
-			logger.Warn().Err(err).Msg("unable to update job status: finish")
-		}
```

Comment:
This smells, I do not remember why I wrote this, finish function should be only called as the very last step therefore it does not matter at which step we are. I removed this and now it simply finishes properly, tho, there is some extra logging above just to inform us in case we finish a reservation that did not execute all steps.